### PR TITLE
Implement statekeeper for tracking DOM changes.

### PR DESCRIPTION
**Pull request by bierik**
Closes https://github.com/4teamwork/bl.web/issues/162

DOM modifications from the simplelayout are not tracked
from the browser as a new page. So we have to trigger
a manual history change.

Tested with 
`http://localhost`,
`http://localhost?key=value`,
`http://localhost?key=value&anohterkey=antohervalue`,
`http://localhost#hash`,
`http://localhost?key=value#hash`,
`http://localhost?key=value&anohterkey=antohervalue#hash`

### CODE REVIEW

File: docs/HISTORY.txt [hunk 1]
```diff
@@ -5,6 +5,9 @@ Changelog
 1.2.2 (unreleased)
 ------------------
 
+- Implement statekeepter to track DOM changes.
```
Comment (jone) [inline docs/HISTORY.txt]: `statekeepter` -> `statekeeper`

**Comment by bierik** [comment]
@jone updated

File: ftw/simplelayout/browser/resources/main.js [hunk 2]
```diff
@@ -2,6 +2,57 @@
...
+    var counter = 0;
+
+    function parseQueryString(queryString) {
+      if(!queryString) {
+        return {};
+      }
+      var params = {}
+      var queries
+      var temp
+      var i
```
Comment (jone) [inline ftw/simplelayout/browser/resources/main.js]: There are some semicolons missing..

File: ftw/simplelayout/browser/resources/main.js [hunk 3]
```diff
@@ -2,6 +2,57 @@
...
+      var l;
+      queries = queryString.split("&");
+      for (i = 0, l = queries.length; i < l; i++) {
+        temp = queries[i].split("=");
+        params[temp[0]] = temp[1];
+      }
+      return params;
+    };
```
Comment (jone) [inline ftw/simplelayout/browser/resources/main.js]: Superfluous trailing semicolon; please use a linter.

File: ftw/simplelayout/browser/resources/main.js [hunk 4]
```diff
@@ -2,6 +2,57 @@
...
+
+    function parseURL(url) {
+      url = url || document.URL;
+      var parser = document.createElement("a");
+      parser.href = url;
+      return parser;
+    }
+
+    function restore() { counter = parseInt(global.localStorage.getItem("sl-state-counter")) || 0; }
```
Comment (jone) [inline ftw/simplelayout/browser/resources/main.js]: please declare radix

**Pull request merged**